src/frontend/metrics/generateMetricsReport.test.ts

```diff
@@ -0,0 +1,85 @@
+import generateMetricsReport from './generateMetricsReport';
+
+describe('generateMetricsReport', () => {
+  const defaultOptions: Parameters<typeof generateMetricsReport>[0] = {
+    packageJson: {version: '1.2.3'},
+    os: 'android',
+    osVersion: 123,
+    screen: {width: 12, height: 34},
+    observations: [
+      // Middle of the Atlantic
+      {lat: 10, lon: -33},
+      // Mexico City
+      {lat: 19.419914, lon: -99.088059},
+      // Machias Seal Island, disputed territory
+      {lat: 44.5, lon: -67.101111},
+      // To be ignored
+      {},
+      {lat: 12},
+      {lon: 34},
+    ],
+  };
+
+  it('can be serialized and deserialized as JSON', () => {
+    const report = generateMetricsReport(defaultOptions);
+    const actual = JSON.parse(JSON.stringify(report));
+    const expected = removeUndefinedEntries(report);
+    expect(actual).toEqual(expected);
+  });
+
+  it('includes a report type', () => {
+    const report = generateMetricsReport(defaultOptions);
+    expect(report.type).toBe('metrics-v1');
+  });
+
+  it('includes the app version', () => {
+    const report = generateMetricsReport(defaultOptions);
+    expect(report.appVersion).toBe('1.2.3');
+  });
```

Comment:
Would this test be more meaningful if we actually pulled the version from the package json?

Reply:
Done in 9b3b7c09f656aec124cbac6844ea6a2992e60358.